sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -346,6 +357,17 @@ export class ServiceBusClient {
       options3
     );
 
+    if (
+      this._connectionContext.config.entityPath &&
+      this._connectionContext.config.entityPath !== entityPath
+    ) {
+      if (entityPath === queueOrTopicName1) {
+        throw new Error(entityPathMisMatchErrors.queue);
+      } else {
+        throw new Error(entityPathMisMatchErrors.topicAndSubscription);
```

Comment:
Does a connection string meant for a topic work for the subscription under it?
If so, we should not throw an error.

Reply:
One may have a SAS policy to just send to the topic. This cannot be used to receive the messages. So, I believe this is still applicable

Reply:
From the portal, a connection string cannot be created for a subscription but is possible for a topic.
If the topic connection string is used to receive messages from the subscription, we should allow it and not throw an error.

Reply:
Tested with SB Explorer, the connection string for a topic obtained from the portal can be used to receive messages from a subscription.

Reply:
Match the entity path and not throw error from the SDK in the following cases
- if there is no entity path in the conn string
- if the entity path in the conn string is `"<topic-name>"` (This is not handled in the current PR)
- if the entity path in the conn string is `"<topic-name>/Subscriptions/<subscription-name>"` (Portal does not allow such conn strings, not sure if this check is needed)

Reply:
Here is the .NET validation - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClient.cs#L566

We just validate that if the topic or queue is in the connection string, the same topic or queue is being passed into CreateX..

Reply:
Nice, that aligns with my comment!
We'll do the same in JS.

Reply:
Great find @HarshaNalluru!

@JoshLove-msft, you seem to be using the below error message which makes sense for Event Hubs where connection string and the entity name were passed to the same method. But in Service Bus, the connection string is passed to the client constructor where as the entity name is passed to the methods on the client. 

> The path to an Service Bus entity may be specified as part of the connection string or as a separate value, but not both.  

I would recommend using a different error message. In JS, am going with

> The queue or topic name provided does not match the EntityPath in the connection string passed to the ServiceBusClient constructor.

Thoughts?

Reply:
> Great find @HarshaNalluru!
> 
> @JoshLove-msft, you seem to be using the below error message which makes sense for Event Hubs where connection string and the entity name were passed to the same method. But in Service Bus, the connection string is passed to the client constructor where as the entity name is passed to the methods on the client.
> 
> > The path to an Service Bus entity may be specified as part of the connection string or as a separate value, but not both.
> 
> I would recommend using a different error message. In JS, am going with
> 
> > The queue or topic name provided does not match the EntityPath in the connection string passed to the ServiceBusClient constructor.
> 
> Thoughts?

Makes sense!

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/14975